File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] prefix = new int[nums.Length];
        int num = 1;
        for(int i = 0; i < prefix.Length; i++){
            prefix[i] = num;
            num = nums[i] * num;
        }

        int[] suffix = new int[nums.Length];
        int index = nums.Length - 1;
        num = 1;
        for(int i = nums.Length - 1; i >= 0; i--){
            suffix[index] = num;
            num = nums[i] * num;
            index--;
        }
        int[] res = new int[nums.Length];

        for(int i = 0; i < nums.Length; i++){
            res[i] = suffix[i] * prefix[i];
        }
        return res;

    }
}
 

File: Data Structures & Algorithms/binary-search/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0; 
        int r = nums.length - 1;

        while(l <= r){
            int middle = ((r - l) / 2) + l;

            if(nums[middle] > target){
                r = middle - 1;
            }else if(nums[middle] < target){
                l = middle + 1;
            }else{
                return middle;
            }
        }
        return -1;
    }
}
